bridge-ffi/src/lib.rs

```diff
@@ -319,17 +319,15 @@ pub extern "C" fn tmprl_register_worker(
             }
         };
     let user_data = UserDataHandle(user_data);
-    core.tokio_runtime.clone().spawn(async move {
-        match core.register_worker(wrappers::WorkerConfig(req)).await {
-            Ok(()) => unsafe {
-                callback(user_data.into(), &*DEFAULT_REGISTER_WORKER_RESPONSE_BYTES);
-            },
-            Err(err) => {
-                let resp = bridge::RegisterWorkerResponse { error: Some(err) };
-                unsafe { callback(user_data.into(), core.encode_proto(&resp).into_raw()) };
-            }
+    match core.register_worker(wrappers::WorkerConfig(req)) {
+        Ok(()) => unsafe {
+            callback(user_data.into(), &*DEFAULT_REGISTER_WORKER_RESPONSE_BYTES);
```

Comment:
Should this be turned into a sync api and drop the callback completely? Do we even have any of those for the FFI layer (don't remember)?

Reply:
I dunno, good question but we can change in another PR if we want. Keeping it callback based is sorta nice in that it's consistent, but I could go either way. @cretz ?